website/docs/advanced-concepts/multiple-environments/multiple-environments-first-draft.md

```diff
@@ -0,0 +1,63 @@
+
+**What environments are available on Appsmith?**
+
+Appsmith provides two environments out of the box - staging and production. Staging environment configurations should ideally point to data sources which are used for development, QA or user acceptance testing. Production environment configurations should point to data sources which have live, end-user facing data.
+
+**How can I configure my environments?**
+
+Environments can be configured by going to the create/edit datasource page. By default, all of the configuration values for a data source are shared across staging and production. If you want to have different values for staging and production, you can do so by going to the “manage environments” page. On this page, you can configure different values for any of the available fields for staging and production.
+
+**How can I toggle between environments?**
+
+In edit mode, you have the option to toggle between staging and production environments. You can find the toggle on the bottom right hand side of the screen.
+
+Toggling between environments in edit mode will have no impact on the deployed instance of the app. However, toggling will impact all other users of the app who are using it in edit mode.
+
+**Which environment is applied for production deployments?**
+
+The deployed version of the app always uses the production environment. However, if the user has requisite permissions, they will be able to “apply” the staging environment configuration in view mode.
+
+**Access controls**
+
+You can specify access controls for the staging and production environments. For each of these environments, you can specify if the user can read the environment, update the environment and apply the environment configuration in edit/view modes.
+
+
+----------------------------------------------------
+# Multiple Environments Outline
+--
+
+## Introduction
+
+Information about multiple environments
+
+* **What are Multiple Environments?**
+
+* **Why do we need Multiple Environments?**
+
+* **What Environments are available on Appsmith?**  
+Explain the differences between the production and staging environments.
+
+  
+## How to Configure an Environment  
+This will be the main section of the document, where we will explain how users can configure environments.  
+### Configure databases  
+In this section, we will discuss how users can configure databases, like PostgreSQL,  
+MongoDB, MySQL, etc. We will add videos, screenshots, and steps to setup the environment.
+
+  
+### Configure APIs  
+In this section, we will discuss how users can configure APIs, like REST, GraphQL, Google Sheets, etc. We will add videos/ screenshots, and steps to setup the environment.
+
+  
+## Switch environments  
+In this section, we will explain how users can switch between different environments. and discuss a few questions like: Which environment is applied for production deployments?
+
+## User's access to environments  
```

Comment:
@sbalaji1192 this will be out only with RBAC, right? 

@harshilp24 Keep this heading  **Access control** as in the first draft. We might have to link to the section in the RBAC docs that gives this info when it is available

Reply:
Keeping it unresolved for @sbalaji1192 to confirm about RBAC

Reply:
@danciaclara did you mean to tag balaji gopinath?

Reply:
Oops so sorry. yes, meant to tag @sribalajig

Reply:
yes, @sribalajig could you please review the outline for multiple environments?